specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/apimgateways.json

```diff
@@ -0,0 +1,1015 @@
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/GatewayIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/IfMatchRequiredParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Gateway was successfully deleted."
+          },
+          "204": {
+            "description": "The Gateway was successfully deleted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/gateways/{gatewayId}/keys": {
+      "post": {
+        "tags": [
+          "GatewayKey"
+        ],
+        "operationId": "Gateway_Keys",
+        "description": "Retrieves gateway keys.",
+        "x-ms-examples": {
+          "ApiManagementGatewayKeys": {
+            "$ref": "./examples/ApiManagementGatewayKeys.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/GatewayIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response body contains the specified Gateway keys.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/GatewayKeysContract"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/gateways/{gatewayId}/regeneratePrimaryKey": {
```

Comment:
>regeneratePrimaryKey [](start = 149, length = 20)

you shouldn't need two different APIs here.  Please follow Storage's model for key regeneration since it is what the majority of users will be familiar with (and it only involves one API):  [https://docs.microsoft.com/en-us/rest/api/storagerp/storageaccounts/regeneratekey](https://docs.microsoft.com/en-us/rest/api/storagerp/storageaccounts/regeneratekey)